lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -167,6 +175,14 @@ public ParquetReader(
         checkArgument(blocks.size() == firstRowsOfBlocks.size(), "elements of firstRowsOfBlocks must correspond to blocks");
 
         this.columnIndexStore = columnIndexStore;
+        this.writeValidation = requireNonNull(writeValidation, "writeValidation is null");
+        validateWrite(
+                validation -> fileCreatedBy.equals(Optional.of(validation.getCreatedBy())),
```

Comment:
Since trino writer always writes a createdBy, and validation is only for trino writer, it should always be non-empty when validation is present.
This is already getting verified because `createdBy` is not an optional in `ParquetWriteValidation`.